### PR TITLE
[CELEBORN-588] Remove test conf's category

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Remove test conf's category to avoid make user confused.

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @pan3793

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 1]
```diff
@@ -1353,7 +1353,7 @@ object CelebornConf extends Logging {
 
   val TEST_RETRY_REVIVE: ConfigEntry[Boolean] =
     buildConf("celeborn.test.retryRevive")
-      .categories("client")
+      .categories("test")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1356]:
```suggestion
      .internal
      .categories("client", "test")
```

Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala]: Done.

**Comment by AngersZhuuuu** [comment]
ping @pan3793

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**